### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by satyaog**
Fix `state_dict(self)` not returning an independent `dict`
Append fidelity dimension into Trial after conversion from Ax space
Fix Ax space conversion to follow Ax picky expectations

### CODE REVIEW

File: src/orion/algo/ax/axoptimizer.py [hunk 1]
```diff
@@ -115,6 +116,7 @@ def suggest(self, num):
                 parameters, _ = client.get_next_trial()
 
                 new_trial = format_trials.dict_to_trial(parameters, self.space)
+                new_trial = self.format_trial(new_trial)
```
Comment (bouthilx) [inline src/orion/algo/ax/axoptimizer.py:119]: Was it failing without `self.format_trial` here?

Comment (satyaog) [reply src/orion/algo/ax/axoptimizer.py:120]: No but wasn't sure if it could with in other context?

File: tests/integration_test.py [hunk 2]
```diff
@@ -10,6 +10,7 @@ class TestAxOptimizer(BaseAlgoTests):
     """Test suite for algorithm AxOptimizer"""
 
     algo_name = "axoptimizer"
+    max_trials = 50
```
Comment (satyaog) [inline tests/integration_test.py:13]: Otherwise `test_is_done_max_trials`  fails for taking too much time

File: src/orion/algo/ax/axoptimizer.py [hunk 3]
```diff
@@ -49,7 +50,7 @@ def __init__(self,
     def state_dict(self):
         """Return a state dict that can be used to reset the state of the algorithm."""
         state_dict = super(AxOptimizer, self).state_dict
-        state_dict["_client_state"] = self._client_state
+        state_dict["_client_state"] = copy.deepcopy(self._client_state)
```
Comment (satyaog) [inline src/orion/algo/ax/axoptimizer.py:53]:
```suggestion
        # AxClient.to_json_snapshot() is more like the current internal state of the client than independent a snapshot
        state_dict["_client_state"] = copy.deepcopy(self._client_state)
```

**Pull request merged**